posthog/ai/utils.py

```diff
@@ -172,7 +172,31 @@ def merge_system_prompt(kwargs: Dict[str, Any], provider: str):
         from posthog.ai.gemini.gemini_converter import format_gemini_input
 
         contents = kwargs.get("contents", [])
-        return format_gemini_input(contents)
+        formatted_messages = format_gemini_input(contents)
+        
+        # Check if system instruction is provided in config parameter
```

Comment:
If this is provider specific formatting, it should live in that provider's converter file.